great_expectations/data_context/types/base.py

```diff
@@ -2568,7 +2551,7 @@ def __init__(  # noqa: PLR0913
         name: Optional[str] = None,
         expectation_suite_name: Optional[str] = None,
         batch_request: Optional[dict] = None,
-        action_list: Optional[Sequence[ActionDict]] = None,
+        action_list: Optional[Sequence[dict]] = None,
```

Comment:
Removing `CheckpointConfig` is a separate effort so just keeping this untyped for now